src/compiler/crystal/syntax/location.cr

```diff
@@ -9,6 +9,32 @@ class Crystal::Location
   def initialize(@filename : String | VirtualFile | Nil, @line_number : Int32, @column_number : Int32)
   end
 
+  # Returns a location from a string representation. Used by compiler tools like
+  # `context` and `implementations`.
+  def self.parse(cursor : String, *, expand : Bool = false) : self
+    file_and_line, colon, col = cursor.rpartition(':')
+    raise ArgumentError.new "cursor location must be file:line:column" if colon.empty?
+
+    file, colon, line = file_and_line.rpartition(':')
+    raise ArgumentError.new "cursor location must be file:line:column" if colon.empty?
+
```

Comment:
```suggestion
    file_and_line, _, col = cursor.rpartition(':')
    file, _, line = file_and_line.rpartition(':')
```
I'm pretty sure this can be simplified as such and the result of this code will be the same in all cases.